Add option to skip validation when dealing with schema registry (#97)
logstash-plugins/logstash-integration-kafka: docs/input-kafka.asciidoc

Use either the Schema Registry config option or the
<<plugins-{type}s-{plugin}-value_deserializer_class>> config option, but not both.

[id="plugins-{type}s-{plugin}-validate_schema_registry"]
===== `schema_registry_validation`

>   * Value can be either of: `auto`, `skip`

Reviewer: 👍🏼 I like the `auto`/`skip` here. It conveys the complexity a lot better than a `true`/`false` would, and similarly doesn't convey promises that we will validate it (if it isn't set, or we can't validate) or that we won't ever do so (e.g., when we start using it and it doesn't provide what we need).

  * Default value is `"auto"`

NOTE: Under most circumstances, the default setting of `auto` should not need to be changed.

> When using the schema registry, by default a validation process to check connectivity and
> the validity of the schema registry takes place during plugin registration. Under certain
> circumstances when trying to validate an authenticated schema registry, this process may
> fail, and cause the plugin to crash. This setting allows this validation to be skipped during
> registration.

Reviewer 2: ```suggestion
When using the schema registry, the plugin checks connectivity and validates the schema registry by default. 
In some circumstances, this process may fail when it tries to validate an authenticated schema registry, causing the plugin to crash.
This setting allows the plugin to skip validation during registration.
```

Reviewer 2: Suggested rewording for consideration.

Author: ```suggestion
When using the schema registry, by default the plugin checks connectivity and validates the schema registry, during plugin registration, before events are processed.
In some circumstances, this process may fail when it tries to validate an authenticated schema registry, causing the plugin to crash.
This setting allows the plugin to skip validation during registration, which allows the plugin to continue and events to be processed. Note that an incorrectly configured schema registry will still stop the plugin from processing events.
```

Author: @karenzone How about this?

Reviewer 2: I'm a fan! Let's go with it.

Reviewer 2: The missing reference problem has been resolved. Wasn't sure it you wanted me to commit your suggestion and then approve, or if you had a different workflow in mind. It should be an easy signoff whenever you're ready for final review.
